From 7797cfa21d42b8706fc6d02bb4106ad8c3335b4e Mon Sep 17 00:00:00 2001
From: Egor Pasko <pasko@google.com>
Date: Tue, 18 May 2010 15:43:14 +0200
Subject: [PATCH 08/12] [gcc] Don't output "add %gs:0x0,%eax" as part of a TLS access

This instruction is currently not accepted by the validator,
although the validator is being overly cautious here.

BUG: http://code.google.com/p/nativeclient/issues/detail?id=263
---
 gcc/gcc/config/i386/i386.md |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gcc/gcc/config/i386/i386.md b/gcc/gcc/config/i386/i386.md
index f8cca8c..152151e 100644
--- a/gcc/gcc/config/i386/i386.md
+++ b/gcc/gcc/config/i386/i386.md
@@ -16559,7 +16559,7 @@
 	(plus:SI (unspec:SI [(const_int 0)] UNSPEC_TP)
 		 (match_operand:SI 1 "register_operand" "0")))
    (clobber (reg:CC FLAGS_REG))]
-  "!TARGET_64BIT"
+  "!TARGET_64BIT && !flag_control_integrity"
   "add{l}\t{%%gs:0, %0|%0, DWORD PTR gs:0}"
   [(set_attr "type" "alu")
    (set_attr "modrm" "0")
-- 
1.7.0.1

